### PR TITLE
[privateuse1] _refs.masked_fill support privateuse1 when value.device.type is cpu

### LINKED ISSUE #124693: masked_fill supports PrivateUse1, when value.device.type is cpu
### 🐛 Describe the bug

We added a device as privateuse1. I'm having trouble running hf_DistilBert using benchmark. The error message is as follows
```
RuntimeError: Expected `value` to be on same device as `a`
```
The corresponding code is
```
@register_decomposition(aten.masked_fill)
@out_wrapper()
def masked_fill(a: TensorLikeType, mask: TensorLikeType, value: TensorOrNumberLikeType):
    python_type = utils.dtype_to_type(a.dtype)
    if isinstance(value, Number):
        value_type = type(value)
    else:
        # NOTE: Could not use value = item(value) as it resulted in
        # RuntimeError: Cannot cast FakeTensor(cpu) to number
        value_ndim = value.ndim
        torch._check(
            value_ndim == 0,
            lambda: f"only supports a 0-dimensional value tensor, but got tensor with {value_ndim} dimension",
        )
        # `masked_fill` allows cpu scalar to be moved to cuda and xpu but not otherwise.
        is_cpu_scalar = a.device.type in ["cuda", "xpu"] and value.device.type == "cpu"
        torch._check(
            is_cpu_scalar or value.device == a.device,
            lambda: "Expected `value` to be on same device as `a`",
        )
        value_type = utils.dtype_to_type(value.dtype)
```

https://github.com/pytorch/pytorch/blob/34bce27f0d12bf7226b37dfe365660aad456701a/torch/_refs/__init__.py#L5555

When a.device.type is `torch._C._get_privateuse1_backend_name()`（Not cuda or xpu）, and value.device.type is 'cpu', I expect 'is_cpu_scalar' is True, but because a.device.type does not equal to 'cuda' or 'xpu', causing 'torch.check( is_cpu_scalar or value.device == a.device,)' failed.

Modifying the code as follows can make the model pass, but I don't know if there will be problems or other restrictions.
```
 is_cpu_scalar = a.device.type in ["cuda", "xpu", torch._C._get_privateuse1_backend_name()`] and value.device.type == "cpu"
```

### Versions

pytorch commit id:3df1b3b0ade3214b9030224667b187bb1d712b1f

### PR DESCRIPTION
**Pull request by markc-614**
_refs.masked_fill support privateuse1 when value.device.type is cpu.

1. maybe I should consider whether this modification meets the expectations of other privateuse1 devices,
2. add TestCase

Fixes #124693

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Thanks for the PR,
Can you add a test for this by any chance?

**Comment by bmeoaountiful** [comment]
> Thanks for the PR, Can you add a test for this by any chance?

@albanD OK, added. please help to review this, thank you

File: test/test_decomp.py [hunk 1]
```diff
@@ -682,6 +682,24 @@ def forward(self, x_1, start_1):
     convert_element_type = torch.ops.prims.convert_element_type.default(add, torch.float32);  add = None
     return convert_element_type""")
 
+    def test_masked_fill(self, device):
+        from torch.fx.experimental.proxy_tensor import make_fx
+        device = device.rstrip(":0123456789")
+        if device not in ["xpu", "cuda", torch._C._get_privateuse1_backend_name()]:
```
Comment (albanD) [inline test/test_decomp.py:688]:
```suggestion
        if torch.device(device).type not in ["xpu", "cuda", torch._C._get_privateuse1_backend_name()]:
```

**Review by albanD — APPROVED** [review]
Small nit on the test but sounds good otherwise, thanks!

File: test/test_decomp.py (hunk 1 above)
Comment (bmeoaountiful) [reply test/test_decomp.py]: ok, updated.

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by albanD** [comment]
@pytorchbot merge -i

These were broken in trunk

**Pull request closed without merging**